Dockerfile.scratch

```diff
@@ -1,42 +1,52 @@
 # Build stage
+# syntax = docker/dockerfile:1.4.2@sha256:443aab4ca21183e069e7d8b2dc68006594f40bddf1b15bbd83f5137bd93e80e2
 ARG goversion
-FROM golang:${goversion}-alpine as builder
-RUN apk add build-base git mercurial ca-certificates
-RUN apk add --update gcc musl-dev
-ADD go.mod /spire/go.mod
-RUN cd /spire && go mod download
-ADD . /spire
+FROM --platform=${BUILDPLATFORM} golang:${goversion}-buster as builder
```

Comment:
Do we need to switch base image to buster?

Can we adjust this to alpine to keep things as much as possible same as original?